lib/migration.js

```diff
@@ -11,8 +11,34 @@
 
 module.exports = Migration;
 
+var isGeneratorFn = require('is-generator').fn;
+
+var co;
+try {
+  co = require('co');
+} catch (err) {
+  // ignore
+}
+
 function Migration(title, up, down) {
   this.title = title;
-  this.up = up;
-  this.down = down;
-}
\ No newline at end of file
+  this.up = migrationFunction(up);
+  this.down = migrationFunction(down);
+}
+
+function migrationFunction(originalFn) {
+  var fn = originalFn;
+
+  if (isGeneratorFn(originalFn)) {
+    if (!co) {
+      throw new Error("Generator migrations can only be used if the 'co' library is included. " +
+        "Run 'npm install --save co' to use generator migrations.");
+    }
+
+    fn = function (next) {
+      return co(originalFn).then(next.bind(null, null), next);
```

Comment:
To be clear, you're referring to when next is null, undefined or not a function or some similar error rather than a function throws when invoked, correct?